### PR TITLE
OV-369: Implement create avatar flow

### LINKED ISSUE #369: TASK: Implement the transition between components in create-avatar page
### What Task?
Implement Transition Between Components in create-avatar page

#### Summary:
Modify the "Next Step" buttons in the instructions, upload, and consent components to enable seamless navigation between them. Ensure the back button in the header and the progress bar accurately reflect the user's current position in the workflow.

#### Details:

1. **Next Step Button Functionality:**
   - Update the "Next Step" buttons in the instructions, upload, and consent components so that, when clicked, the next component in the sequence is rendered.

2. **Back Button in Header:**
   - Modify the back button in the header to allow the user to navigate to the previous component in the flow.

3. **Stepper Updates:**
   - Ensure the stepper accurately reflects the user's current position, updating as they move between components (e.g., instructions, upload, and consent).

This task requires https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/pull/70

### PR DESCRIPTION
**Pull request by stefano-lacorazza**
Implemented flow between, instructions, upload and consent components

https://github.com/user-attachments/assets/17674b5b-dd45-4841-b810-cccb8dfa4f30

### CODE REVIEW

File: frontend/src/bundles/create-avatar/pages/create-avatar.tsx [hunk 1]
```diff
@@ -1,13 +1,67 @@
...
+    useMemo,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
 
-import { Instruction } from '../components/components.js';
+import { Consent, Instruction } from '../components/components.js';
 import styles from './styles.module.css';
 
 const CreateAvatar: React.FC = () => {
+    const steps = useMemo(() => ['Instructions', 'Upload', 'Consent'], []);
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/create-avatar/pages/create-avatar.tsx]: can we create object for this values because you reuse them a lot

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/stepper/stepper.tsx [hunk 2]
```diff
@@ -19,9 +19,10 @@ import styles from './styles.module.css';
 type Properties = {
     steps: string[];
     currentStep: string;
+    onClickBack?: () => void;
 };
 
-const Stepper: React.FC<Properties> = ({ steps, currentStep }) => {
+const Stepper: React.FC<Properties> = ({ steps, currentStep, onClickBack }) => {
     const activeStepIndex = steps.indexOf(currentStep);
     const progressPercent = (activeStepIndex / (steps.length - 1)) * 100;
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/common/components/stepper/stepper.tsx]: Magic numbers

**Review by sofiia-trokhymchuk — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/stepper/stepper.tsx [hunk 3]
```diff
@@ -24,7 +24,9 @@ type Properties = {
 
 const Stepper: React.FC<Properties> = ({ steps, currentStep, onClickBack }) => {
     const activeStepIndex = steps.indexOf(currentStep);
-    const progressPercent = (activeStepIndex / (steps.length - 1)) * 100;
+    const MAX_PERCENT = 100;
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/common/components/stepper/stepper.tsx]: It is better to declare the constants outside of the component. Not necessary in an external file but outside of the component

**Review by sofiia-trokhymchuk — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/create-avatar/components/instruction/instruction.tsx [hunk 4]
```diff
@@ -22,7 +22,11 @@ const avoidListItems = [
     'Hand gestures above the chest or pointing gestures',
 ];
 
-const Instruction: React.FC = () => {
+type InstructionProperties = {
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/create-avatar/components/instruction/instruction.tsx]:
```suggestion
type Properties = {
```

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**